### DISCUSSION TITLE
Use @pm2/io (or async_hooks)?

### DISCUSSION THREAD

**Discussion by kyleawayan** [vercel/next.js#17611] [Help]
I'm trying to make [pm2 show custom metrics](https://pm2.io/docs//plus/guide/custom-metrics/) for all of [next.js' vitals and custom metrics](https://nextjs.org/docs/advanced-features/measuring-performance).

I put the `export function reportWebVitals(metric)` function in my `_app.js`, and I plan to use `@pm2/io` to send those stats back to the pm2 web interface. But, when I try to use the `@pm2/io` module, I get an `Module not found: Can't resolve 'async_hooks'` error.

I looked it up, and it says that [`webpack` is doing something with the `async_hooks` module causing it not to work](https://github.com/elastic/apm-agent-nodejs/issues/1075#issuecomment-494587021). I have no idea what to do or put in `next.config.js`, even though I saw they have a way to change the `webpack` config settings.

What's the correct way of getting `async_hooks` to work?
Thanks.

**Comment by stephen-dahl** [comment]
just wrap your import in if(!process.browser) any code in that if statement wont be included in the browser build
